nucypher/blockchain/eth/sol/source/contracts/proxy/Upgradeable.sol

```diff
@@ -19,11 +19,17 @@ contract Upgradeable is Ownable {
     * However the storage layout is specified here in the implementing contracts.
     **/
     address public target;
+
     /**
     * @dev Previous contract address (if available). Used for rollback
     **/
     address public previousTarget;
 
+    /**
+    * @dev Secret hash to proof that user owns previous version of a contract
+    **/
+    bytes32 public secretHash;
```

Comment:
Heh - This reads funny.  @szotov - Can you clarify what is being exposed here?

Reply:
Maybe previous version of contract is not the best name
Who owns contract that already deployed also should know secret (proof of owning) to use it in upgrade